File: android/app/src/main/java/com/celalbasaran/stripmate/data/model/DailyPrompt.kt
package com.celalbasaran.stripmate.data.model

import com.google.firebase.firestore.DocumentSnapshot
import java.util.Date

data class DailyPrompt(
    val id: String = "",
    val promptText: String = "",
    val emoji: String = "",
    val category: PromptCategory = PromptCategory.RANDOM,
    val activeDate: Date = Date(),
    val isCompletedToday: Boolean = false
) {
    fun toMap(): Map<String, Any?> = buildMap {
        put("id", id)
        put("promptText", promptText)
        put("emoji", emoji)
        put("category", category.value)
        put("activeDate", com.google.firebase.Timestamp(activeDate))
        put("isCompletedToday", isCompletedToday)
    }

    companion object {
        fun fromDocument(doc: DocumentSnapshot): DailyPrompt? {
            if (!doc.exists()) return null
            val categoryStr = doc.getString("category") ?: "random"
            return DailyPrompt(
                id = doc.id,
                promptText = doc.getString("promptText") ?: "",
                emoji = doc.getString("emoji") ?: "",
                category = PromptCategory.fromString(categoryStr),
                activeDate = doc.getTimestamp("activeDate")?.toDate() ?: Date(),
                isCompletedToday = doc.getBoolean("isCompletedToday") ?: false
            )
        }

        val PROMPT_LIBRARY: List<PromptEntry> = listOf(
            // Selfie
            PromptEntry("bugün nasıl görünüyorsun? hadi bir selfie!", "\uD83E\uDD33", PromptCategory.SELFIE),
            PromptEntry("en doğal halini görmek istiyoruz, filtre yok!", "\uD83E\uDE9E", PromptCategory.SELFIE),
            PromptEntry("bugünkü enerjini yüzünden okuyalım", "\uD83D\uDE01", PromptCategory.SELFIE),
            PromptEntry("en sevdiğin eşyanla bir selfie çeker misin?", "\u2764\uFE0F", PromptCategory.SELFIE),

            // Mood
            PromptEntry("günaydın! sabahın ilk anı nasıl görünüyor?", "\uD83C\uDF05", PromptCategory.MOOD),
            PromptEntry("bugün kendini nasıl hissediyorsun? tek kareyle anlat", "\uD83D\uDCAD", PromptCategory.MOOD),
            PromptEntry("bugün seni gülümseten küçük şey ne oldu?", "\uD83D\uDE0A", PromptCategory.MOOD),
            PromptEntry("şu anki modunu en iyi anlatan kare hangisi?", "\u2728", PromptCategory.MOOD),

            // Place
            PromptEntry("şu an tam olarak neredesin, göster bakalım", "\uD83D\uDCCD", PromptCategory.PLACE),
            PromptEntry("evdeki en rahat köşeni merak ediyoruz", "\uD83C\uDFE0", PromptCategory.PLACE),
            PromptEntry("pencerenden ne görünüyor şu an?", "\uD83E\uDE9F", PromptCategory.PLACE),
            PromptEntry("bugün en çok vakit geçirdiğin yer neresi?", "\uD83D\uDCBB", PromptCategory.PLACE),

            // Food
            PromptEntry("bugün ne yiyorsun, bize de göster!", "\uD83C\uDF7D\uFE0F", PromptCategory.FOOD),
            PromptEntry("kahven mi çayın mı? hadi görelim", "\u2615", PromptCategory.FOOD),
            PromptEntry("bugünkü atıştırmalığın ne, merak ettik", "\uD83C\uDF7F", PromptCategory.FOOD),
            PromptEntry("mutfakta bir şeyler mi pişiriyorsun? göster!", "\uD83D\uDC68\u200D\uD83C\uDF73", PromptCategory.FOOD),

            // Creative
            PromptEntry("etrafına bak, sence en güzel detay hangisi?", "\uD83C\uDFA8", PromptCategory.CREATIVE),
            PromptEntry("en renkli şeyi bul ve çek, renk avı!", "\uD83C\uDF08", PromptCategory.CREATIVE),
            PromptEntry("telefonu ters çevir, baş aşağı bir kare çek!", "\uD83D\uDE43", PromptCategory.CREATIVE),
            PromptEntry("bir gölge ya da yansıma yakala", "\uD83C\uDF17", PromptCategory.CREATIVE),
            PromptEntry("bir şeyin çok yakınından çek, ne olduğunu biz tahmin edelim", "\uD83D\uDD0D", PromptCategory.CREATIVE),
            PromptEntry("etrafında yüze benzeyen bir şey var mı?", "\uD83D\uDC40", PromptCategory.CREATIVE),

            // Social
            PromptEntry("yanındaki en sevdiğin insanla bir kare!", "\uD83D\uDC6F", PromptCategory.SOCIAL),
            PromptEntry("şu an kiminle birliktesin? göster!", "\uD83E\uDEC2", PromptCategory.SOCIAL),
            PromptEntry("bugün gördüğün en tatlı canlı kim?", "\uD83D\uDC3E", PromptCategory.SOCIAL),
            PromptEntry("birlikte olduğun arkadaşlarınla grup fotoğrafı!", "\uD83D\uDCF8", PromptCategory.SOCIAL),

            // Nature
            PromptEntry("başını kaldır, gökyüzü nasıl görünüyor?", "\uD83C\uDF24\uFE0F", PromptCategory.NATURE),
            PromptEntry("etrafında yeşil bir şey bul ve çek", "\uD83C\uDF3F", PromptCategory.NATURE),
            PromptEntry("bugün hava nasıl? bir kareyle anlat", "\uD83C\uDF21\uFE0F", PromptCategory.NATURE),
            PromptEntry("yakınındaki bir çiçek veya bitki var mı?", "\uD83C\uDF38", PromptCategory.NATURE),

            // Random
            PromptEntry("ayağındakilere bak, bugün ne giydin?", "\uD83D\uDC5F", PromptCategory.RANDOM),
            PromptEntry("son aldığın şey neydi? göster bakalım", "\uD83D\uDECD\uFE0F", PromptCategory.RANDOM),
            PromptEntry("telefonunun ekranında şu an ne var?", "\uD83D\uDCF1", PromptCategory.RANDOM),
            PromptEntry("etrafında mavi bir şey bul!", "\uD83D\uDC99", PromptCategory.RANDOM),
            PromptEntry("bugünkü kombinin nasıl?", "\uD83D\uDC57", PromptCategory.RANDOM),
            PromptEntry("yanındaki en rastgele objeyi çek", "\uD83C\uDFB2", PromptCategory.RANDOM),
            PromptEntry("gurur duyduğun bir şeyi göster bize", "\uD83C\uDFC6", PromptCategory.RANDOM),
            PromptEntry("sahip olduğun en eski eşya hangisi?", "\uD83D\uDD70\uFE0F", PromptCategory.RANDOM),
            PromptEntry("cebinde veya çantanda ne var?", "\uD83D\uDC5C", PromptCategory.RANDOM),
            PromptEntry("bugünkü planların neler, göster!", "\uD83D\uDCDD", PromptCategory.RANDOM),

            // Extra variety
            PromptEntry("ayna karşısında bir selfie zamanı!", "\uD83E\uDE9E", PromptCategory.SELFIE),
            PromptEntry("sabah kalktığında ilk gördüğün şey ne?", "\u23F0", PromptCategory.MOOD),
            PromptEntry("kapından dışarı çıkınca ilk ne görüyorsun?", "\uD83D\uDEAA", PromptCategory.PLACE),
            PromptEntry("en sevdiğin bardak veya kupayı göster", "\uD83C\uDF75", PromptCategory.FOOD),
            PromptEntry("simetrik bir kare yakalayabilir misin?", "\u2696\uFE0F", PromptCategory.CREATIVE),
            PromptEntry("bugün gördüğün en güzel davranış neydi?", "\uD83D\uDC9B", PromptCategory.SOCIAL),
            PromptEntry("gün batımını veya doğumunu yakaladın mı?", "\uD83C\uDF07", PromptCategory.NATURE),
            PromptEntry("etrafında kırmızı bir şey bul!", "\u2764\uFE0F", PromptCategory.RANDOM),
            PromptEntry("şu an ne okuyorsun veya ne izliyorsun?", "\uD83D\uDCD6", PromptCategory.RANDOM),
            PromptEntry("ellerinle bir şey yapıyorsan göster!", "\uD83E\uDD32", PromptCategory.CREATIVE),
            PromptEntry("bugün gününü güzelleştiren şey ne oldu?", "\uD83C\uDF1F", PromptCategory.MOOD),
            PromptEntry("en çok sevdiğin köşeyi göster", "\uD83D\uDECB\uFE0F", PromptCategory.PLACE),
            PromptEntry("bir dokunun yakın çekimini yap", "\uD83E\uDDF1", PromptCategory.CREATIVE),
            PromptEntry("çocukluğundan kalan bir eşyan var mı?", "\uD83E\uDDF8", PromptCategory.RANDOM),
            PromptEntry("bu gece gökyüzü nasıl görünüyor?", "\uD83C\uDF19", PromptCategory.NATURE),
            PromptEntry("çok minik bir şey bul ve çek", "\uD83D\uDC1C", PromptCategory.CREATIVE),
            PromptEntry("siyah-beyaz çekilmeyi hak eden bir kare bul", "\uD83D\uDDA4", PromptCategory.CREATIVE),
            PromptEntry("ayaklarına ve zeminine bak, ne görüyorsun?", "\uD83D\uDC63", PromptCategory.RANDOM),
            PromptEntry("şu an kulaklığından ne çalıyor?", "\uD83C\uDFB5", PromptCategory.RANDOM),
            PromptEntry("ilginç bir kapı veya pencere yakala", "\uD83D\uDEAA", PromptCategory.CREATIVE)
        )
    }
}

data class PromptEntry(
    val text: String,
    val emoji: String,
    val category: PromptCategory
)

enum class PromptCategory(val value: String, val displayName: String, val icon: String) {
    SELFIE("selfie", "Selfie", "person"),
    MOOD("mood", "Ruh Hali", "sentiment_satisfied"),
    PLACE("place", "Mekan", "place"),
    FOOD("food", "Yemek", "restaurant"),
    CREATIVE("creative", "Yaratıcı", "brush"),
    SOCIAL("social", "Sosyal", "group"),
    NATURE("nature", "Doğa", "eco"),
    RANDOM("random", "Rastgele", "casino");

    companion object {
        fun fromString(value: String): PromptCategory =
            entries.firstOrNull { it.value == value } ?: RANDOM
    }
}

File: StripMate/Models/DailyPrompt.swift
import Foundation

/// Represents a daily photo challenge/prompt shown to users.
/// Stored in Firestore at: `daily_prompts/{dateString}` (e.g. "2026-03-03")
public struct DailyPrompt: Identifiable, Codable, Sendable {
    /// Date string in "yyyy-MM-dd" format
    public let id: String
    /// The challenge text shown to the user
    public let promptText: String
    /// Localization key for the prompt (if localized prompts are needed later)
    public let promptKey: String
    /// Emoji icon for visual flair
    public let emoji: String
    /// Category for grouping/theming
    public let category: PromptCategory
    /// Timestamp when this prompt becomes active
    public let activeDate: Date

    public enum PromptCategory: String, Codable, Sendable, CaseIterable {
        case selfie = "selfie"
        case mood = "mood"
        case place = "place"
        case food = "food"
        case creative = "creative"
        case social = "social"
        case nature = "nature"
        case random = "random"

        public var displayName: String {
            switch self {
            case .selfie: return String(localized: "Selfie")
            case .mood: return String(localized: "Ruh Hali")
            case .place: return String(localized: "Mekan")
            case .food: return String(localized: "Yemek")
            case .creative: return String(localized: "Yaratıcı")
            case .social: return String(localized: "Sosyal")
            case .nature: return String(localized: "Doğa")
            case .random: return String(localized: "Rastgele")
            }
        }

        public var icon: String {
            switch self {
            case .selfie: return "person.fill"
            case .mood: return "face.smiling"
            case .place: return "mappin.and.ellipse"
            case .food: return "fork.knife"
            case .creative: return "paintbrush.fill"
            case .social: return "person.2.fill"
            case .nature: return "leaf.fill"
            case .random: return "dice.fill"
            }
        }
    }
}

// MARK: - Built-in Prompt Library

public extension DailyPrompt {
    /// A curated list of 60 prompts that rotate. Cloud Function picks one per day.
    static let promptLibrary: [(text: String, emoji: String, category: PromptCategory)] = [
        // Selfie — samimi, eğlenceli
        ("bugün nasıl görünüyorsun? hadi bir selfie!", "person.fill", .selfie),
        ("en doğal halini görmek istiyoruz, filtre yok!", "person.crop.square", .selfie),
        ("bugünkü enerjini yüzünden okuyalım", "face.smiling.inverse", .selfie),
        ("en sevdiğin eşyanla bir selfie çeker misin?", "heart.fill", .selfie),

        // Ruh Hali — düşündüren, sıcak
        ("günaydın! sabahın ilk anı nasıl görünüyor?", "sunrise.fill", .mood),
        ("bugün kendini nasıl hissediyorsun? tek kareyle anlat", "thought.bubble", .mood),
        ("bugün seni gülümseten küçük şey ne oldu?", "face.smiling", .mood),
        ("şu anki modunu en iyi anlatan kare hangisi?", "sparkles", .mood),

        // Mekan — keşfe teşvik
        ("şu an tam olarak neredesin, göster bakalım", "mappin", .place),
        ("evdeki en rahat köşeni merak ediyoruz", "house.fill", .place),
        ("pencerenden ne görünüyor şu an?", "window.horizontal", .place),
        ("bugün en çok vakit geçirdiğin yer neresi?", "desktopcomputer", .place),

        // Yemek — sıcak, gündelik
        ("bugün ne yiyorsun, bize de göster!", "fork.knife", .food),
        ("kahven mi çayın mı? hadi görelim", "cup.and.saucer.fill", .food),
        ("bugünkü atıştırmalığın ne, merak ettik", "popcorn.fill", .food),
        ("mutfakta bir şeyler mi pişiriyorsun? göster!", "frying.pan.fill", .food),

        // Yaratıcı — ilham veren
        ("etrafına bak, sence en güzel detay hangisi?", "paintpalette.fill", .creative),
        ("en renkli şeyi bul ve çek, renk avı!", "rainbow", .creative),
        ("telefonu ters çevir, baş aşağı bir kare çek!", "arrow.uturn.down", .creative),
        ("bir gölge ya da yansıma yakala", "circle.lefthalf.filled", .creative),
        ("bir şeyin çok yakınından çek, ne olduğunu biz tahmin edelim", "magnifyingglass", .creative),
        ("etrafında yüze benzeyen bir şey var mı?", "eye.fill", .creative),

        // Sosyal — arkadaşça
        ("yanındaki en sevdiğin insanla bir kare!", "person.2.fill", .social),
        ("şu an kiminle birliktesin? göster!", "figure.2", .social),
        ("bugün gördüğün en tatlı canlı kim?", "pawprint.fill", .social),
        ("birlikte olduğun arkadaşlarınla grup fotoğrafı!", "camera.fill", .social),

        // Doğa — gözlem
        ("başını kaldır, gökyüzü nasıl görünüyor?", "cloud.sun.fill", .nature),
        ("etrafında yeşil bir şey bul ve çek", "leaf.fill", .nature),
        ("bugün hava nasıl? bir kareyle anlat", "thermometer.medium", .nature),
        ("yakınındaki bir çiçek veya bitki var mı?", "camera.macro", .nature),

        // Rastgele — eğlenceli, şaşırtıcı
        ("ayağındakilere bak, bugün ne giydin?", "shoeprint.fill", .random),
        ("son aldığın şey neydi? göster bakalım", "bag.fill", .random),
        ("telefonunun ekranında şu an ne var?", "iphone", .random),
        ("etrafında mavi bir şey bul!", "drop.fill", .random),
        ("bugünkü kombinin nasıl?", "tshirt.fill", .random),
        ("yanındaki en rastgele objeyi çek", "dice.fill", .random),
        ("gurur duyduğun bir şeyi göster bize", "trophy.fill", .random),
        ("sahip olduğun en eski eşya hangisi?", "clock.fill", .random),
        ("cebinde veya çantanda ne var?", "bag.fill", .random),
        ("bugünkü planların neler, göster!", "list.clipboard", .random),

        // Ekstra çeşitlilik
        ("ayna karşısında bir selfie zamanı!", "person.crop.square", .selfie),
        ("sabah kalktığında ilk gördüğün şey ne?", "alarm.fill", .mood),
        ("kapından dışarı çıkınca ilk ne görüyorsun?", "door.left.hand.open", .place),
        ("en sevdiğin bardak veya kupayı göster", "mug.fill", .food),
        ("simetrik bir kare yakalayabilir misin?", "square.split.2x2", .creative),
        ("bugün gördüğün en güzel davranış neydi?", "hand.thumbsup.fill", .social),
        ("gün batımını veya doğumunu yakaladın mı?", "sunset.fill", .nature),
        ("etrafında kırmızı bir şey bul!", "heart.fill", .random),
        ("şu an ne okuyorsun veya ne izliyorsun?", "book.fill", .random),
        ("ellerinle bir şey yapıyorsan göster!", "hand.raised.fill", .creative),
        ("bugün gününü güzelleştiren şey ne oldu?", "star.fill", .mood),
        ("en çok sevdiğin köşeyi göster", "sofa.fill", .place),
        ("bir dokunun yakın çekimini yap", "square.grid.3x3.fill", .creative),
        ("çocukluğundan kalan bir eşyan var mı?", "teddybear.fill", .random),
        ("bu gece gökyüzü nasıl görünüyor?", "moon.fill", .nature),
        ("çok minik bir şey bul ve çek", "ant.fill", .creative),
        ("siyah-beyaz çekilmeyi hak eden bir kare bul", "circle.lefthalf.filled", .creative),
        ("ayaklarına ve zeminine bak, ne görüyorsun?", "figure.walk", .random),
        ("şu an kulaklığından ne çalıyor?", "music.note", .random),
        ("ilginç bir kapı veya pencere yakala", "door.left.hand.open", .creative),
    ]
}
